inc/compat/themes/compat-theme-smart-home.php

```diff
@@ -0,0 +1,159 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Compatibility with theme: Smart Home (by Fuel Themes).
+ */
+class FluidCheckout_ThemeCompat_SmartHome extends FluidCheckout {
+
+	/**
+	 * __construct function.
+	 */
+	public function __construct() {
+		$this->hooks();
+	}
+
+
+
+	/**
+	 * Initialize hooks.
+	 */
+	public function hooks() {
+		// Very late hooks
+		add_action( 'wp', array( $this, 'very_late_hooks' ), 100 );
+
+		// Checkout page template
+		add_filter( 'template_include', array( $this, 'checkout_page_template' ), 110 );
+
+		// Template file loader
+		add_filter( 'woocommerce_locate_template', array( $this, 'locate_template_checkout_page_template' ), 100, 3 );
+
+		// Container class
+		add_filter( 'fc_add_container_class', '__return_false', 10 );
+		add_filter( 'fc_content_section_class', array( $this, 'change_fc_content_section_class' ), 10 );
+
+		// Remove theme elements from checkout page
+		remove_action( 'woocommerce_checkout_before_customer_details', 'thb_checkout_before_customer_details', 5 );
+		remove_action( 'woocommerce_checkout_after_customer_details', 'thb_checkout_after_customer_details', 30 );
+		remove_action( 'woocommerce_checkout_after_order_review', 'thb_checkout_after_order_review', 30 );
+
+		// Sticky elements
+		add_filter( 'fc_checkout_progress_bar_attributes', array( $this, 'change_sticky_elements_relative_header' ), 20 );
+		add_filter( 'fc_checkout_sidebar_attributes', array( $this, 'change_sticky_elements_relative_header' ), 20 );
+	}
+
+	/**
+	 * Add or remove very late hooks.
+	 */
+	public function very_late_hooks() {
+		if ( FluidCheckout_Steps::instance()->is_checkout_page_or_fragment() ) {
+			// Prevent theme from replacing `select2` field assets on checkout page
+			remove_action( 'wp_enqueue_scripts', 'thb_woocommerce_scripts_styles', 10001 );
+		}
```

Comment:
- Medium: Move grouped hooks which need an `if` statement to an additional `..._hooks()` function to avoid higher code complexity, and use "Bail" statements inside the additional `..._hooks()` function for the necessary checks.